src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+the transaction that service clients use to invoke it.
+
+The method accepts `AddVehicle` class: an auto-generated class from our
+protobuf definition. It also accepts the `TransactionContext` allowing
+the transaction to access the database.
+
+The transaction implementation uses the service data schema `MySchema`
+to access its data.
+
+Note that the transaction throws an `ExecutionException` in case of
+a precondition failure: in this case, an attempt to add a vehicle with
+an existing ID.
+
+<!--
+todo: Shall we add the ids to the example? If so, how do we split ids for
+different transactions?
+Or just use the literals?
+-->
+
+Compile the code:
+
+```shell
+mvn compile
+```
+
+#### Transfer the Ownership Transaction
+
+Let's add the second transaction which will change the owner of a vehicle
+in the registry. It needs as its arguments the ID of the vehicle;
+and a new owner.
+
+Navigate to `transactions.proto` and add a message with the arguments:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-messages/src/main/proto/example/vehicle/transactions.proto)
+block:ChangeOwner
+<!--/codeinclude-->
+
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+Then navigate to the service `MyService` and add an implementation with
+appropriate constants:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-change-owner
+<!--/codeinclude-->
+
+This transaction is similar to the first.
+
+Notice how an update of the `owner` field in an existing `Vehicle` value
+is performed. It creates a builder from the existing object with
+`Vehicle.newBuilder(templateVehicle)` method, updates the field, and builds
+a new object.
+
+Compile the code:
+
+```shell
+mvn compile
+```
+
+#### Service Constructor
+
+Finally, we will add a _service constructor_ — a special type of a transaction,
+that is invoked once when the service is instantiated. We will use it
+to populate our registry with some test data.
+
+It is implemented as the `Service#initialize` method. By default, it has
+an empty implementation in the interface, hence it is not yet present
+in `MyService`.
+
+Override the `Service#initialize` with the following implementation:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-initialize
+<!--/codeinclude-->
+
+Note that this method delegates to the `addVehicle` transaction method
+we have added earlier.
+
+!!! success
+    In this section we have learned how to implement operations modifying
+    the blockchain state: transactions and the service constructor.
+    <!-- todo: "There are more operations of such type: ..." — what is
+    the canonical reference on the topic? Shall we add one? -->
```

Comment:
I think you're looking for the *Hooks* section in the *Services* article. IMO, it makes sense to add a link to it.

Reply:
Now that I think of it, it seems redundant, because we expect the reader to follow the tutorial, not to read on everything the Service abstraction provides (all the possible hooks). WDYT?

Reply:
There is a separate question though on what to suggest at the end of the tutorial :-)